modelci/types/models/mlmodel.py

```diff
@@ -7,45 +7,61 @@
 """
 import getpass
 import os
+import gridfs
 from datetime import datetime
 from pathlib import Path
 from typing import Union, Optional, Dict, List, Any
 
 from bson import ObjectId
-from gridfs import GridOut
 from pydantic import BaseModel, FilePath, DirectoryPath, PositiveInt, Field, root_validator
 
 from .common import Metric, IOShape, Framework, Engine, Task, ModelStatus, Status, PydanticObjectId, \
     named_enum_json_encoder
 from .pattern import as_form
 from ...hub.utils import parse_path_plain, generate_path_plain
+from modelci.config import db_settings
+from modelci.experimental.mongo_client import MongoClient
+
+_db = MongoClient()[db_settings.mongo_db]
+_fs = gridfs.GridFS(_db)
 
 
 class Weight(BaseModel):
     """TODO: Only works for MLModelIn"""
 
-    __slots__ = ('file',)
+    __slots__ = ('file', 'grid_out')
 
     __root__: Optional[PydanticObjectId]
 
     def __init__(self, __root__):
         if isinstance(__root__, Path):
             object.__setattr__(self, 'file', FilePath.validate(__root__))
+            object.__setattr__(self, 'grid_out', None)
             __root__ = None
 
-        self._grid_out: Optional[GridOut]
+        if isinstance(__root__, ObjectId):
+            if _fs.exists(__root__):
+                object.__setattr__(self, 'file', None)
+                object.__setattr__(self, 'grid_out', _fs.get(__root__))
```

Comment:
What does `grid_out` means?

Reply:
We should hide GridFS knowledge from the user side.

Reply:
`Weight` class has a `_grid_out` properties previously, it's GridOut instance of target model file

Reply:
> We should hide GridFS knowledge from the user side.

Should we make `grid_out` private attribute?

Reply:
better gridfs_out as grid is misleading.